modules/trustxBidAdapter.js

```diff
@@ -130,30 +161,57 @@ function _getBidFromResponse(respItem) {
   return respItem && respItem.bid && respItem.bid[0];
 }
 
-function _addBidResponse(serverBid, bidsMap, priceType, bidResponses) {
+function _addBidResponse(serverBid, bidsMap, priceType, bidResponses, remainingServerBids) {
   if (!serverBid) return;
   let errorMessage;
   if (!serverBid.auid) errorMessage = LOG_ERROR_MESS.noAuid + JSON.stringify(serverBid);
   if (!serverBid.adm) errorMessage = LOG_ERROR_MESS.noAdm + JSON.stringify(serverBid);
   else {
+    const ignoreSizes = !remainingServerBids;
```

Comment:
@PWyrembak This makes it possible that TrustX will return a `bidResponse` with a different size for an AdUnit which does not request that size?

For instance if I request two adUnits,

```javascript
  {
    code: 'Test-Slot-1',
    mediaTypes: {
      banner: {
        sizes: [
        [728, 90]
        ],
      }
    },
    bids: [{
          bidder: "trustx",
          params: {
              uid: '44',
              priceType: 'gross' // by default is 'net'
          }
      }]
  },
  {
    code: 'Test-Slot-2',
    mediaTypes: {
      banner: {
        sizes: [
        [300, 250], [300, 600]
        ],
      }
    },
    bids: [{
          bidder: "trustx",
          params: {
              uid: 45,
              priceType: 'gross'
          }
      }]
  }
```

Then I can see TrustX bid on the 728x90 slot but it uses a size from the other adUnit.

Reply:
![image](https://user-images.githubusercontent.com/13972794/53035873-507b3280-342b-11e9-80b2-ba4c51713c67.png)

I am not sure this is desirable?

What is the reason for doing this?

If TrustX is flexible about the creative sizes it returns, then why not mark the bidResponse object with a valid size from the bidRequest?

Reply:
This can also lead to confusion within the analytics adapters and downstream GPT reporting possibly if I am not mistaken.

Reply:
@robertrmartinez We have updated our bid adapter logic in order to ignore bids which don't match respective slot sizes. Please review.